Introduce messages fence mechanism for Key_Shared subscription (#6977)
apache/pulsar: site2/docs/reference-configuration.md

|statusFilePath|  Path for the file used to determine the rotation status for the broker when responding to service discovery health checks ||
|preferLaterVersions| If true, (and ModularLoadManagerImpl is being used), the load manager will attempt to use only brokers running the latest software version (to minimize impact to bundles)  |false|
|maxNumPartitionsPerPartitionedTopic|Max number of partitions per partitioned topic. Use 0 or negative number to disable the check|0|

> |subscriptionKeySharedEnable| Enable Key_Shared subscription |true|

Reviewer: ```suggestion
|subscriptionKeySharedEnable| Enable Key_Shared subscription. |true|
```

> |maxFencedMessagesForKeySharedSubscription| Max fenced messages for Key_Shared subscription. Messages for the new consumer will be fenced until the previous messages are acknowledged |10000|

Reviewer: ```suggestion
|maxFencedMessagesForKeySharedSubscription| Set the maximum number of fenced messages for Key_Shared subscription. Messages for a new consumer are fenced until the previous messages are acknowledged. |10000|
```
